### PR TITLE
Bug 1882479: erroring when replacement not found

### PR DESCRIPTION
**Pull request by gallettilance**
**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/registry/populator_test.go [hunk 1]
```diff
@@ -631,6 +631,37 @@ func TestListBundles(t *testing.T) {
...
+	}
+	add := map[image.Reference]string{
+		image.SimpleReference("quay.io/test/etcd.0.9.2"):        "../../bundles/etcd.0.9.2",
+		image.SimpleReference("quay.io/test/prometheus.0.22.2"): "../../bundles/prometheus.0.22.2",
+	}
+	expectedErr := errors.NewAggregate([]error{
+		fmt.Errorf("Invalid bundle %s, bundle specifies a non-existent replacement %s", "etcdoperator.v0.9.2", "etcdoperator.v0.9.0"),
+		fmt.Errorf("Invalid bundle %s, bundle specifies a non-existent replacement %s", "prometheusoperator.0.22.2", "prometheusoperator.0.15.0"),
+	})
+	require.ElementsMatch(t, expectedErr, populate(add))
```
Comment (njhale) [inline pkg/registry/populator_test.go:664]: I don't think `TestListBundles` is the appropriate spec for this. It probably needs its own.

File: pkg/registry/replacesgraphloader.go [hunk 2]
```diff
@@ -15,11 +15,21 @@ func (r *ReplacesGraphLoader) CanAdd(bundle *Bundle, graph *Package) (bool, erro
 		return false, fmt.Errorf("Invalid content, unable to parse bundle")
 	}
 
+	// adding the first bundle in the graph
+	if replaces == "" {
+		return true, nil
+	}
+	csvName := bundle.Name
```
Comment (njhale) [inline pkg/registry/replacesgraphloader.go:22]: nit: `csvName` is only used once, so we should probably omit the local variable in favor of `bundle.Name`.

File: pkg/registry/replacesgraphloader.go [hunk 3]
```diff
@@ -15,11 +15,21 @@ func (r *ReplacesGraphLoader) CanAdd(bundle *Bundle, graph *Package) (bool, erro
 		return false, fmt.Errorf("Invalid content, unable to parse bundle")
 	}
 
+	// adding the first bundle in the graph
+	if replaces == "" {
```
Comment (njhale) [inline pkg/registry/replacesgraphloader.go:19]: this is duplicated

Comment (gallettilance) [reply pkg/registry/replacesgraphloader.go:19]: nice catch - must have happened when I reverted this file...

File: pkg/registry/populator.go [hunk 4]
```diff
@@ -302,15 +302,11 @@ func (i *DirectoryPopulator) getNextReplacesImagesToAdd(imagesToAdd []*ImageInpu
 		// no new images can be added, the current iteration aggregates all the
 		// errors that describe invalid bundles
 		if pkgFoundImages == 0 && pkgRemainingImages > 0 {
```
Comment (njhale) [inline pkg/registry/populator.go:304]: nit: you could always use the lengths of `remainingImages` and `foundImages` instead of these counters

Comment (gallettilance) [reply pkg/registry/populator.go:304]: +1 just didn't want to change too much about the add when implementing the overwrite fix

**Review by ecordell — APPROVED** [review]
/lgtm

**Pull request merged**